### PR TITLE
feat: Stop using submitter client in SDK

### PR DESCRIPTION
**Pull request by ddiakiteaneo**

### CODE REVIEW

**Review by fdenefaneo — CHANGES_REQUESTED** [review]
Some comments are just questions. Others are questions/requests for change, depending on their suitability.

File: ArmoniK.SDK.Client/src/SessionServiceImpl.cpp [hunk 1]
```diff
@@ -70,40 +72,65 @@ const std::string &SessionServiceImpl::getSession() const { return session; }
...
+
+    task_creations.push_back(std::move(creation));
   }
+
   auto reply = channel_pool.WithChannel([&](auto channel) {
-    return armonik::api::client::SubmitterClient(armonik::api::grpc::v1::submitter::Submitter::NewStub(channel))
-        .create_tasks_async(session, static_cast<armonik::api::grpc::v1::TaskOptions>(task_options), definitions)
-        .get();
+    return armonik::api::client::TasksClient(armonik::api::grpc::v1::tasks::Tasks::NewStub(channel))
+        .submit_tasks(session, task_creations, static_cast<armonik::api::grpc::v1::TaskOptions>(task_options));
```
[the remark this answers is not in the document]
Comment (ddiakiteaneo) [reply ArmoniK.SDK.Client/src/SessionServiceImpl.cpp:111]: No, the task creation is synchronous now !

File: ArmoniK.SDK.Client/src/SessionServiceImpl.cpp [hunk 2]
```diff
@@ -70,40 +72,65 @@ const std::string &SessionServiceImpl::getSession() const { return session; }
...
-  definitions.reserve(task_requests.size());
 
-  // Creates the needed result ids
-  auto result_ids = generate_result_ids(task_requests.size());
+  std::vector<armonik::api::common::TaskCreation> task_creations;
+  task_creations.reserve(task_requests.size());
 
   for (size_t i = 0; i < task_requests.size(); i++) {
     armonik::api::grpc::v1::TaskRequest request;
+    std::map<std::string, std::string> name_payload;
```
[the remark this answers is not in the document]
Comment (fdenefaneo) [reply ArmoniK.SDK.Client/src/SessionServiceImpl.cpp]: OK, hope it is going to be deleted :slightly_smiling_face:

File: ArmoniK.SDK.Client/src/SessionServiceImpl.cpp [hunk 3]
```diff
@@ -70,40 +72,65 @@ const std::string &SessionServiceImpl::getSession() const { return session; }
...
+    });
+
+    // Set payload ID
+    creation.payload_id = std::move(result_payload.second);
+    // One result per task
+    creation.expected_output_keys.push_back(std::move(result_payload.first));
+    creation.data_dependencies.insert(creation.data_dependencies.end(), task_requests[i].data_dependencies.begin(),
+                                      task_requests[i].data_dependencies.end());
+
+    task_creations.push_back(std::move(creation));
```
[the remark this answers is not in the document]
Comment (fdenefaneo) [reply ArmoniK.SDK.Client/src/SessionServiceImpl.cpp]: Oh OK, I did not know this detail.

**Review by fdenefaneo — APPROVED** [review]

**Pull request merged**